build/guide/multi-platform.md

```diff
@@ -0,0 +1,263 @@
+```
+
+The status for the new `container` builder is inactive. That's fine - it's
+because you haven't started using it yet.
+
+## Build using emulation
+
+To run multi-platform builds with Buildx, invoke the `docker buildx build`
+command, and pass it the same arguments as you did to the regular `docker build`
+command before. Only this time, also add:
+
+- `--builder=container` to select the new builder
+- `--platform=linux/amd64,linux/arm/v7,linux/arm64/v8` to build for multiple
+  architectures at once
+
+```console
+$ docker buildx build \
+    --target=binaries \
+    --output=bin \
+    --builder=container \
+    --platform=linux/amd64,linux/arm64,linux/arm/v7 .
+```
+
+This command uses emulation to run the same build four times, once for each
+platform. The build results are exported to a `bin` directory.
+
+```text
+bin
+├── linux_amd64
+│   ├── client
+│   └── server
+├── linux_arm64
+│   ├── client
+│   └── server
+└── linux_arm_v7
+    ├── client
+    └── server
+```
+
+When you build using a builder that supports multi-platform builds, the builder
+runs all of the build steps under emulation for each platform that you specify.
+Effectively forking the build into two concurrent processes.
+
+![Build pipelines using emulation](./images/emulated-builds.png){:.invertible}
+
+There are, however, a few downsides to running multi-platform builds using
+emulation:
+
+- If you tried running the command above, you may have noticed that it took a
+  long time to finish. Emulation can be much slower than native execution for
+  CPU-intensive tasks.
+- Emulation only works when the architecture is supported by the base image
+  you’re using. The example in this guide uses the Alpine Linux version of the
+  `golang` image, which means you can only build Linux images this way, for a
+  limited set of CPU architectures, without having to change the base image.
+
+As an alternative to emulation, the next step explores cross-compilation.
+Cross-compiling makes multi-platform builds much faster and versatile.
+
+## Build using cross-compilation
+
+Using cross-compilation means leveraging the capabilities of a compiler to build
+for multiple platforms, without the need for emulation.
+
+The first thing you'll need to do is pinning the builder to use the node’s
+native architecture as the build platform. This is to prevent emulation. Then,
+from the node's native architecture, the builder cross-compiles the application
+to a number of other target platforms.
+
+### Platform build arguments
+
+This approach involves using a few pre-defined build arguments that you have
+access to in your Docker builds: `BUILDPLATFORM` and `TARGETPLATFORM` (and
+derivatives, like `TARGETOS`). These build arguments reflect the values you pass
+to the `--platform` flag.
+
+For example, if you invoke a build with `--platform=linux/amd64`, then the build
+arguments resolve to:
+
+- `TARGETOS=linux`
+- `TARGETARCH=amd64`
```

Comment:
I think we should add `TARGETPLATFORM` to this list as well.

Reply:
Agree `TARGETPLATFORM` is good to know if in a follow-up we want to introduce users to cross comp with https://github.com/tonistiigi/xx/.